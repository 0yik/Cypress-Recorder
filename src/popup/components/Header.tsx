import * as React from 'react';
import InfoButton from './InfoButton';

export interface HeaderProps {
  toggleInfoDisplay: () => void,
}

export default ({ toggleInfoDisplay }: HeaderProps) => (
  <div>
    Header
    <InfoButton toggleInfoDisplay={toggleInfoDisplay} />
  </div>
);
