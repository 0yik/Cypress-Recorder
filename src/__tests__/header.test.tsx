import * as React from 'react';
import { shallow } from 'enzyme';
import Header from '../popup/components/Header';

import '../setupTests';

describe('Header', () => {
  let wrapper;
  let toggleInfoMock;
  beforeEach(() => {
    toggleInfoMock = jest.fn();
  });
  it('Title should say "Cypress Recorder"', () => {
    wrapper = shallow(<Header shouldInfoDisplay={false} toggleInfoDisplay={toggleInfoMock} />);
    expect(wrapper).toMatchSnapshot();
    const title = wrapper.find('#title');
    expect(title.text()).toBe('Cypress Recorder');
  });
});
