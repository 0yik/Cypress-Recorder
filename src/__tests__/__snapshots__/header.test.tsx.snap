// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Title should say "Cypress Recorder" 1`] = `
<div
  id="header"
>
  <h1
    id="title"
  >
    Cypress Recorder
  </h1>
  <Component
    shouldInfoDisplay={false}
    toggleInfoDisplay={[MockFunction]}
  />
</div>
`;
